src/lexer/TokenKind.ts

```diff
@@ -598,3 +598,47 @@ export const DeclarableTypes = [
     TokenKind.Void,
     TokenKind.Function
 ];
+
+/**
+ * The tokens that might preceed a regex literal
+ */
+export const PreceedingRegexTypes = new Set([
```

Comment:
This is the full list of tokens that can come before a regex literal. If it's not in this list, then a regex literal will not be tokenized even if we encounter a forward slash. I find it safer to create an allowlist rather than a denylist, as it's easier to test, and we won't accidentally break functionality when we forgot to put something in the denylist, whereas only new adopters of regexp literals would encounter issues when something is missing from the allowlist.